### PR TITLE
[NU-1775] Add sortAscBy and reverse functions to collection helper

### PR DESCRIPTION
**Pull request by lukasz-bigorajski**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala [hunk 1]
```diff
@@ -344,4 +369,45 @@ object CollectionUtils {
...
+        fieldName: Any
+    ): ValidatedNel[GenericFunctionTypingError, typing.TypingResult] = {
+      fields.get(fieldName.asInstanceOf[String]) match {
+        case Some(TypedClass(klass, _)) =>
+          if (comparableClass.isAssignableFrom(klass)) {
+            baseTypeClass.copy(params = parametersTypes.withoutValue :: Nil).validNel
+          } else {
+            GenericFunctionTypingError.OtherError(s"Field must implement Comparable interface").invalidNel
+          }
+        case Some(Unknown) =>
```
[the remark this answers is not in the document]
Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:405]: Done

File: utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala [hunk 2]
```diff
@@ -344,4 +369,45 @@ object CollectionUtils {
...
+    }
+
+    private def listResultType(
+        baseTypeClass: TypedClass,
+        parametersTypes: TypedObjectTypingResult,
+        fields: ListMap[String, typing.TypingResult],
+        fieldName: Any
+    ): ValidatedNel[GenericFunctionTypingError, typing.TypingResult] = {
+      fields.get(fieldName.asInstanceOf[String]) match {
+        case Some(TypedClass(klass, _)) =>
```
[the remark this answers is not in the document]
Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:399]: Done

File: utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala [hunk 3]
```diff
@@ -344,4 +369,45 @@ object CollectionUtils {
...
+          if (comparableClass.isAssignableFrom(klass)) {
+            baseTypeClass.copy(params = parametersTypes.withoutValue :: Nil).validNel
+          } else {
+            GenericFunctionTypingError.OtherError(s"Field must implement Comparable interface").invalidNel
+          }
+        case Some(Unknown) =>
+          GenericFunctionTypingError.OtherError(s"Field must implement Comparable interface").invalidNel
```
[the remark this answers is not in the document]
Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:406]: Done

File: utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala [hunk 4]
```diff
@@ -344,4 +369,45 @@ object CollectionUtils {
...
+        case _ => GenericFunctionTypingError.OtherError(s"Record must contain field: $fieldName").invalidNel
```
[the remark this answers is not in the document]
Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:407]: Done

File: utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala [hunk 5]
```diff
@@ -121,6 +123,29 @@ trait CollectionUtils extends HideToString {
...
+  @GenericType(typingFunction = classOf[RecordCollectionSortingTyping])
+  def sortedAscBy(
+      @ParamName("list") list: java.util.Collection[java.util.Map[String, Any]],
+      @ParamName("fieldName") fieldName: String
+  ): java.util.List[java.util.Map[String, Any]] = {
+    list.asScala.toList.sortWith { (firstMap, secondMap) =>
+      (firstMap.get(fieldName), secondMap.get(fieldName)) match {
+        case (a, b) if a != null && b != null =>
+          a.asInstanceOf[Comparable[Any]].compareTo(b.asInstanceOf[Comparable[Any]]) < 0
+        case _ => throw new IllegalStateException("Elements cannot be compared")
```
[the remark this answers is not in the document]
Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:139]: Done

File: utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala [hunk 6]
```diff
@@ -121,6 +123,29 @@ trait CollectionUtils extends HideToString {
     values
   }
 
+  @Documentation(description =
+    "Returns a list of all elements sorted by record field (field cannot be nullable) " +
```
[the remark this answers is not in the document]
Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:127]: Done

[the remark this answers is not in the document]
Comment (mk-software-pl) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala]: is it? I'm not sure I see where is the check

**Review by mk-software-pl — APPROVED** [review]
LGTM

**Pull request merged**